source/extensions/filters/http/cache/cache_policy.h

```diff
@@ -23,82 +24,122 @@ struct CacheEntryUsability {
    * Value to be put in the Age header for cache responses.
    */
   Seconds age = Seconds::max();
+  /**
+   * Remaining freshness lifetime--how long from now until the response is stale. (If the response
+   * is already stale, `ttl` should be negative.)
+   */
+  Seconds ttl = Seconds::max();
+
+  friend bool operator==(const CacheEntryUsability& a, const CacheEntryUsability& b) {
+    return std::tie(a.status, a.age, a.ttl) == std::tie(b.status, b.age, b.ttl);
+  }
+
+  friend bool operator!=(const CacheEntryUsability& a, const CacheEntryUsability& b) {
+    return !(a == b);
+  }
 };
 
-class CachePolicyCallbacks {
-public:
-  virtual ~CachePolicyCallbacks() = default;
+enum class RequestCacheability {
+  // This request is eligible for serving from cache, and for having its response stored.
+  Cacheable,
+  // Don't respond to this request from cache, or store its response into cache.
+  Bypass,
+  // This request is eligible for serving from cache, but its response must not be stored. (E.g.
+  // requests with "Cache-Control: no-store").
```

Comment:
I think Raven asked this already but I missed the answer, but  how does a no-store request get served from cache if the response cannot be stored? Maybe you could add that to the comment here?

Reply:
Expanded explanation. Is that sufficiently clear?